### ISSUE TITLE
Vehicle stops in front of non-existing obstacle on the path

### ISSUE THREAD

**Issue by WJaworskiRobotec** [autowarefoundation/autoware.universe#2213]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

While testing current Autoware in AWSIM for ITRI BusODD scenarios, the following problem was discovered :

The vehicle stops because the obstacle on the path is reported, but when detected objects are visualized, it seems that the objects are not really blocking the path.

- the only detected object is far from the vehicle, and nnot on the path
![Screenshot from 2022-11-03 11-50-28](https://user-images.githubusercontent.com/85557834/199752475-4e30d34b-d0a2-4c0f-99b5-329eb6d975f3.png)

- detected object is on the side of the vehicle, and is not blocking the movement

![Screenshot from 2022-11-03 11-32-13](https://user-images.githubusercontent.com/85557834/199753080-ea2bb522-fca2-4631-ba94-3dcdcdca3ee0.png)
![Screenshot from 2022-11-03 11-32-39](https://user-images.githubusercontent.com/85557834/199752934-f9e10b17-1617-4272-a46e-d6c19367232c.png)
![Screenshot from 2022-11-03 11-33-36](https://user-images.githubusercontent.com/85557834/199752939-7cebbf8d-90ec-4dd5-8ff7-f013c3b622f6.png)

Location of the observed problems:

![Screenshot from 2022-11-07 10-34-49](https://user-images.githubusercontent.com/85557834/200276662-e39a7978-e74a-44e1-8e1a-204dde9d37ee.png)

### Expected behavior

 The vehicle should not stop if there are no objects detected on the path

### Actual behavior

The vehicle stops with no objects on the path

### Steps to reproduce

Run AWSIM binary with Autoware - description to be updated once binary is publicly shared

**Comment by maxime-clem** [comment]
An "obstacle on the path" can also be caused by an obstacle _point_.
Have you checked the pointcloud in rviz ? I am afraid the points from the tree leaves are what is causing your issue.
Here is the related open discussion: https://github.com/orgs/autowarefoundation/discussions/2981

**Comment by shmpwk** [comment]
@WJaworskiRobotec 
This problem may comes from [this PR](https://github.com/autowarefoundation/autoware.universe/pull/1899/files).
Could you change these parameters https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_perception_launch/config/obstacle_segmentation/ground_segmentation/ground_segmentation.param.yaml#L14-L15 ?

before:
```
max_z: 10.7 
min_z: -8.7
```

After
```
max_z: 4.5 # in case 50m continuous slope at 5deg
min_z: -4.5# in case 50m continuous slope at 5deg
```

**Comment by shmpwk** [comment]
cc. @badai-nguyen

**Comment by WJaworskiRobotec** [comment]
@maxime-clem so what is the input to the planner from sensing/perception modules ? I thought it's only list of detection objects, but from your comments I understand that it's a pointcloud after some segmentation/cropping as well, right ?

**Comment by maxime-clem** [comment]
@WJaworskiRobotec You are right that the `obstacle_avoidance_planner` only uses the list of detected objects but some other planning modules uses the obstacle pointcloud (most notably the `obstacle_stop_planner`).
[This page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-architecture/planning/) gives a good overview of which planning nodes uses which inputs.

**Comment by yukkysaito** [comment]
cc @takayuki5168

**Comment by xmfcx** [comment]
https://www.thinkautonomous.ai/blog/occupancy-networks/

**Comment by shmpwk** [comment]
Reflected https://github.com/autowarefoundation/autoware.universe/issues/2213#issuecomment-1303143710  by
https://github.com/tier4/autoware_launch/pull/543
https://github.com/autowarefoundation/autoware.universe/pull/2260

**Comment by lchojnack** [comment]
I've found the root cause of stopping the vehicle presented in the issue, and also in the discussion https://github.com/orgs/autowarefoundation/discussions/2981. I've noticed that in these two situations there are tree branches and tree leaves on the EGO path. The tree branches are higher than the vehicle's height and do not cause danger or reason to stop the vehicle. The node which is responsible for detecting points that are on the path is `obstacle_stop_planner`. The `obstacle_stop_planner` is checking if any point from a point cloud is within the polygon trajectory, and decides to stop the vehicle. Checking points only within a polygon (2d information) and not taking into account the vehicle height could cause stopping the vehicle when an obstacle is above and below the vehicle. There is an example of detecting tree branches above the vehicle in the picture below.
<img width="960" alt="Screenshot_20221121_154255" src="https://user-images.githubusercontent.com/8416175/203089923-d1603dd6-98ec-46e9-ba1c-e732a3c7cf14.png">


My proposition is to extend checking if the point is within **a polyhedron** (3d information) instead of if is within a polygon (2d information). The idea is presented in the picture below.
<img width="960" alt="Screenshot_20221121_154647" src="https://user-images.githubusercontent.com/8416175/203089963-bdb9c1c2-fd58-4245-90ac-2585da1c3726.png">
